### PR TITLE
ci(.circleci): don't prefix all jobs with 'test/e2e'

### PR DESCRIPTION
**Pull request by michaelbeaumont**
The CircleCI UI is horrible and everything after `:` is hidden.

See [before](https://app.circleci.com/pipelines/github/kumahq/kuma/21807/workflows/dfa05249-ea74-4a76-9f62-6fe5901b7d67) vs ~[after](https://app.circleci.com/pipelines/github/kumahq/kuma/21815/workflows/f54ff654-affe-4474-bd22-4c9a82205e96)~ [after](https://app.circleci.com/pipelines/github/kumahq/kuma/21895/workflows/820c2cb7-da85-4e3b-ba46-a1346c783dc6)

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
I'm not against this in spirit just needs to adapt updates on KM.

**Comment by lahabana** [comment]
It's barely more readable though ;) Do you know if GHA have the same truncating behaviour?

**Comment by michaelbeaumont** [comment]
@lahabana do you know if we have any other tooling/scripting anywhere that depends on the CircleCI job names?

**Comment by lahabana** [comment]
@michaelbeaumont this script relies on job names: https://github.com/Kong/kong-mesh/blob/master/tools/dev/sync_ci.sh

File: .circleci/config.yml [hunk 1]
```diff
@@ -224,9 +224,9 @@ jobs:
         type: integer
         default: 1
       target:
-        description: makefile target
+        description: makefile target without test/e2e prefix
         type: string
-        default: "test/e2e"
+        default: ""
```
Comment (lahabana) [inline .circleci/config.yml:229]: should the default be `legacy` or `mix` ? Feels more understandable than ""

Comment (michaelbeaumont) [reply .circleci/config.yml:229]: Not sure, given that it's the `e2e` job I would guess `test/e2e` to be the default target. For a string, `""` is an ok default value. That is, if I don't specify a target at all, I'd expect _some_ default to be run.

**Review by lukidzi — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (michaelbeaumont) [reply .circleci/config.yml:229]: I think we should rename `test/e2e`, I'll do this in another PR

**Pull request merged**